spec/importer_spec.rb

```diff
@@ -0,0 +1,178 @@
+require 'spec_helper'
+require 'ronin/masscan/importer'
+require 'masscan/output_file'
+require 'ronin/db'
+
+RSpec.describe Ronin::Masscan::Importer do
+  let(:output_file) { instance_double('Masscan::OutputFile') }
+
+  describe '.import_file' do
+    let(:path) { '/path/to/file' }
+
+    before do
+      allow(Masscan::OutputFile).to receive(:new).with(path).and_return(output_file)
+    end
+
+    it 'must initialize Masscan::OutputFile and import from it' do
+      expect(subject).to receive(:import).with(output_file)
+
+      subject.import_file(path)
+    end
+  end
+
+  describe '.import' do
+    let(:status_record)  { Masscan::Status.new(protocol: :tcp, port: 80, ip: "1.1.1.1", status: "status", timestamp: Time.now) }
+    let(:banner_record)  { instance_double("Masscan::Banner") }
+    let(:unknown_record) { 'unknown_record' }
+
+    before do
+      allow(output_file).to receive(:each).and_yield(unknown_record)
+      allow(subject).to receive(:import_status)
+    end
+
+    it 'raises NotImplementedError for unknown record types' do
+      expect { |b|
+        subject.import(output_file, &b)
+      }.to raise_error(NotImplementedError)
+    end
+
+    context 'when a block is given' do
+      before do
+        allow(output_file).to receive(:each).and_yield(status_record)
+      end
+
+      it 'must import status' do
+        expect(subject).to receive(:import_status)
+
+        subject.import(output_file) { |f| f }
+      end
+    end
+
+    context 'when no block is given' do
+      it 'must return an enumerator' do
+        expect(subject).to receive(:enum_for)
+
+        subject.import(output_file)
+      end
+    end
+  end
+
+  describe '.import_status' do
+    let(:status) { instance_double('Masscan::Status') }
+    let(:ip)     { "1.1.1.1" }
+
+    context 'when Masscan::Status status is :open' do
+      context 'when Masscan::Status protocol is :icmp' do
+        before do
+          allow(status).to receive(:status).and_return(:open)
+          allow(status).to receive(:protocol).and_return(:icmp)
+          allow(status).to receive(:ip).and_return(ip)
+        end
```

Comment:
`Masscan::Status` is a `Struct` class, so we could manually initialize it with the values we want. Or we could pull the first one out of the `Masscan::OutputFile` object using a scan file in `spec/fixtures/`.